lib/friendly_shipping/services/usps_international/parse_rate_response.rb

```diff
@@ -0,0 +1,86 @@
+# frozen_string_literal: true
+
+require 'friendly_shipping/services/usps/parse_xml_response'
+require 'friendly_shipping/services/usps_international/parse_package_rate'
+
+module FriendlyShipping
+  module Services
+    class UspsInternational
+      class ParseRateResponse
+        class BoxNotFoundError < StandardError; end
+
+        class << self
+          # Parse a response from USPS' international rating API
+          #
+          # @param [FriendlyShipping::Request] request The request that was used to obtain this Response
+          # @param [FriendlyShipping::Response] response The response that USPS returned
+          # @param [Physical::Shipment] shipment The shipment object we're trying to get results for
+          # @param [FriendlyShipping::Services::UspsInternational::RateEstimateOptions] options The options we sent with this request
+          # @return [Result<ApiResult<Array<FriendlyShipping::Rate>>>] When successfully parsing, an array of rates in a Success Monad.
+          def call(request:, response:, shipment:, options:)
+            # Filter out error responses and directly return a failure
+            parsing_result = FriendlyShipping::Services::Usps::ParseXMLResponse.call(
+              request: request,
+              response: response,
+              expected_root_tag: 'IntlRateV2Response'
+            )
+            parsing_result.fmap do |xml|
+              # Get all the possible rates for each package
+              rates_by_package = rates_from_response_node(xml, shipment, options)
+
+              rates = SHIPPING_METHODS.map do |shipping_method|
+                # For every package ...
+                matching_rates = rates_by_package.map do |package, package_rates|
+                  # ... choose the rate that matches the shipping method for this package
+                  package_rates.select { |r| r.shipping_method == shipping_method }.first
+                end.compact # Some shipping rates are not available for every shipping method.
```

Comment:
returning them all. I'm expecting caller to sort out. We could find best by the shipping method for the package? My experience there is only ever one rate per package per shipping method...

Reply:
Returning them all is fine.